tests/console/validate_fs_table.pm

```diff
@@ -0,0 +1,65 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Validation of filesystem table type, filesystem partitioning, reading and writting in data or root partitions, check of unpartitioned disk space. 
+# Maintainer: Sofia Syrianidou <ssyrianidou@suse.com>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use scheduler 'get_test_suite_data';
+use Test::Assert ':all';
+
+sub run {
+
+    select_console "root-console";
+    my $test_data = get_test_suite_data();
+    if (defined $test_data->{table_type}) {
+        # Validate that the partition table type is the expected one.
+        my $table_type = script_output("parted -l | sed -n -e 4p | awk {'print \$3'}");
+        assert_equals($test_data->{table_type}, $table_type, "Partition table type does not correspont to the expected one.");
+    }
+    foreach my $disk (@{$test_data->{disks}}) {
+        foreach my $partition (@{$disk->{partitions}}) {
+            if ($_->{role} eq 'swap') {
+                record_info("Check $_->{role}", "Verify the '$_->{role}' partition was created and the filesystem is '$partition->{formatting_options}->{filesystem}'");
+                # Validate that the partition is created.
+                assert_script_run("lsblk -n | grep SWAP", fail_message => "Partition SWAP was not created.");
+                # Validate that the filesystem type is the expected one.
+                my $fstype = script_output("cat /etc/fstab | grep swap | awk '{print \$3}'");
+                assert_equals($partition->{formatting_options}->{filesystem}, $fstype, "File system on '$_->{role}' partition does not correspond to the expected one.");
+            } elsif (($_->{role} eq 'data') || ($partition->{mounting_options}->{mount_point} eq '/')) {
```

Comment:
you could adjust test data to your needs so you can avoid some of those `if/else/elsif/else` conditions, for example adding a key to test data with `validate_rw : 1` to know which partition you wish to validate read/write operation. In that way you might end up with more homogeneous loop which offer some flags inside, same idea than with `$disk->{allowed_unpartitioned}` It would be more readable, but I mean, it might not be a big difference, so up to you if you want leave it as it is.